### ISSUE TITLE
dockerd.exe --register-service should set a service restart action

### ISSUE THREAD

**Issue by PatrickLang** [docker/docker#27696]
**Description**

`dockerd.exe --register-service` is used to create the "Docker" service and set it to automatically start when a Windows system boots up. This was done as an alternative to using something like NSSM and uses the service control manager built into Windows. 

One key feature is that it can restart services automatically, but its not configured by default. My ask is that we enable that by default to:
- Restart the service after 60 seconds on the first or second failure
- Restart the failure counter after a day

Infinite restarts can spend too much CPU cycles in the case of a crash, so I don't think that should be the default.

**Steps to reproduce the issue:**
1. `dockerd.exe --register-service`
2. `net start Docker`
3. `get-process -name dockerd | stop-process -Force`
4. Wait ~1 minute
5. `docker info`
6. `Get-Service Docker`

**Describe the results you received:**

`docker info` fails:
```none
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.25/info: open //./pipe/docker_engine: The system cannot find the file specified.```

```none
 Get-Service Docker

Status   Name               DisplayName
------   ----               -----------
Stopped  Docker             Docker Engine
```

**Describe the results you expected:**
`Get-Service Docker` should show running

`docker info` shouldn't give an error

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64
```

**Output of `docker info`:**

```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 17
Server Version: 1.12.2-cs2-ws-beta
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: nat null overlay
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.321.amd64fre.rs1_release_inmarket.161004-2338)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 7 GiB
Name: plang-ws16-1
ID: 6Z4K:IBTG:Q3H3:YGUR:63I6:A6CT:GPF5:G3CN:COTV:63Q2:F7G5:ACBO
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
No dependency - this repros on any Windows machine, physical or virtual

**Comment by jhowardmsft** [comment]
#dibs

**Comment by jhowardmsft** [comment]
Fixed by #27700
